### PR TITLE
Add Ghost actions

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Adds Ghost actions

## Why It's Good For The Game

QOL

## Changelog
:cl:
add: New observer actionss
/:cl:

----

image

### CODE REVIEW

**Comment by Rohesie** [comment]
Very nice. This also opens the possibility of more actions, such as join as the various available roles and whatnot.

File: code/datums/actions/__defines.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#define XACT_USE_INCAP			(1 << 0) // ignore incapacitated
```
Comment (DominikPanic) [inline code/datums/actions/__defines.dm]: Move it into the _DEFINES folder `actions.dm`

File: code/datums/actions/skill.dm [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/datum/action/skill
+	var/skill_name
+	var/skill_min
+
+/datum/action/skill/should_show()
+	return can_use_action()
+
+/datum/action/skill/can_use_action()
+	var/mob/living/carbon/human/human = owner
+	return istype(human) && human.mind && human.mind.cm_skills && human.mind.cm_skills.vars[skill_name] >= skill_min
```
Comment (DominikPanic) [inline code/datums/actions/skill.dm]:
```suggestion
	return istype(human) human.mind?.cm_skills && human.mind.cm_skills.vars[skill_name] >= skill_min
```

File: code/datums/actions/item_action.dm [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	var/old_plane = I.plane
+	I.layer = ABOVE_HUD_LAYER
+	I.plane = ABOVE_HUD_PLANE
+	button.overlays += I
+	I.layer = old_layer
+	I.plane = old_plane
+
+
+/datum/action/item_action/toggle/New(Target)
+	..()
```
Comment (DominikPanic) [inline code/datums/actions/item_action.dm]:
```suggestion
	. = ..()
```

File: code/datums/actions/item_action.dm [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		holder = target
+	holder_item = holder
+	holder_item.actions += src
+	name = "Use [target]"
+	button.name = name
+
+/datum/action/item_action/Destroy()
+	holder_item.actions -= src
+	holder_item = null
+	..()
```
Comment (DominikPanic) [inline code/datums/actions/item_action.dm]:
```suggestion
	return ..()
```

File: code/datums/actions/item_action.dm [hunk 5]
```diff
@@ -0,0 +1,48 @@
+
+/datum/action/item_action
+	name = "Use item"
+	var/obj/item/holder_item //the item that has this action in its list of actions. Is not necessarily the target
+							//e.g. gun attachment action: target = attachment, holder = gun.
+
+/datum/action/item_action/New(Target, obj/item/holder)
+	..()
```
Comment (DominikPanic) [inline code/datums/actions/item_action.dm]:
```suggestion
	. = ..()
```

**Pull request merged**